pandas/tseries/index.py

```diff
@@ -77,16 +77,19 @@ def f(self):
 
             result = tslib.get_start_end_field(values, field, self.freqstr,
                                                month_kw)
+            result = self._maybe_mask_results(result, convert='float64')
+
```

Comment:
About `is_leap_year`, was done like this in purpose in https://github.com/pandas-dev/pandas/pull/13739, citing @sinhrks "pd.NaT.is_leap_year results in False, as I think users want bool array."

But, this does not seem very consistent with other `is_` methods .. (but I would keep this for another issues/PR)